### ISSUE TITLE
[java] UnusedReturnValue: False positive for calls made on Mockito.verify(mock)

### ISSUE THREAD

**Issue by estekhin** [pmd/pmd#7068]
**Affects PMD Version:** 
7.27.0

**Rule:**
UnusedReturnValue: <https://docs.pmd-code.org/pmd-doc-7.27.0/pmd_rules_java_errorprone.html#unusedreturnvalue>

**Description:**

UnusedReturnValue reports some verification calls to methods called on `Mockito.verify(mock)`.

```java
private static void assertSomethingOnGrpcStub(io.grpc.stub.AbstractStub<?> mock) {
    var captor = ArgumentCaptor.forClass(CallCredentials.class);
    // Rule:UnusedReturnValue Priority:3 Result of 'io.grpc.stub.AbstractStub.withCallCredentials()' is ignored..
    Mockito.verify(mock).withCallCredentials(captor.capture());
}

```

**Expected outcome:**

UnusedReturnValue should not report anything that comes after `Mockito.verify(mock)`.
